system/src/system_sleep.cpp

```diff
@@ -406,3 +383,119 @@ int system_sleep_ext(const hal_sleep_config_t* config, hal_wakeup_source_base_t*
 }
 
 #endif // HAL_PLATFORM_SLEEP_2_0
+
+/**
+ * Wraps the actual implementation, which has to return a value as part of the threaded implementation.
+ */
+int system_sleep_pin(uint16_t wakeUpPin, uint16_t edgeTriggerMode, long seconds, uint32_t param, void* reserved)
+{
+    // Cancel current connection attempt to unblock the system thread
+    network_connect_cancel(0, 1, 0, 0);
+    InterruptMode m = (InterruptMode)edgeTriggerMode;
+    return system_sleep_pin_impl(&wakeUpPin, 1, &m, 1, seconds, param, reserved);
+}
+
+int system_sleep(Spark_Sleep_TypeDef sleepMode, long seconds, uint32_t param, void* reserved)
```

Comment:
Let's think of a test to validate that we haven't broken anything by changing the internal implementation.

Reply:
Is the test to validate if the API is buildable with different arguments, i.e. API test, or to validate if this function behaves as expected?

Reply:
I think we already have API tests, if not we need to add/adjust them. However, I'm talking more about an on-device test that can verify that the device behaves as expected and exactly the same as before this change.